### PR TITLE
feat(internal): add step load and loadlogs functions

### PR DESCRIPTION
**Pull request by jbrockopp**
Part of the effort for https://github.com/go-vela/community/issues/54

This adds a new `go-vela/pkg-executor/internal/step` package.

In this package are a `step.Load()` and `step.LoadLogs()` functions.

The intention of these functions are to load the step or step logs from the client 👍

This benefits us by reducing the non-test LOC in the executor codebase:

https://github.com/go-vela/pkg-executor/blob/bb0ff301cea6c54c6c33964c2ebb960c563389e3/internal/step/load.go#L15-L31

https://github.com/go-vela/pkg-executor/blob/bb0ff301cea6c54c6c33964c2ebb960c563389e3/internal/step/load.go#L33-L49

### CODE REVIEW

File: internal/step/load_test.go [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+import (
+	"reflect"
+	"sync"
+	"testing"
+
+	"github.com/go-vela/types/library"
+	"github.com/go-vela/types/pipeline"
+)
+
+func TestStep_Load(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (jbrockopp) [reply internal/step/load_test.go:16]: Ignoring this 👍

File: internal/step/load_test.go [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+			t.Errorf("Load returned err: %v", err)
+		}
+
+		if !reflect.DeepEqual(got, test.want) {
+			t.Errorf("Load is %v, want %v", got, test.want)
+		}
+	}
+}
+
+func TestStep_LoadLogs(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (jbrockopp) [reply internal/step/load_test.go:83]: Ignoring this 👍

File: executor/linux/secret.go [hunk 3]
```diff
@@ -108,40 +109,44 @@ func (s *secretSvc) destroy(ctx context.Context, ctn *pipeline.Container) error
...
 	if err != nil {
 		// create the step from the container
-		step = new(library.Step)
-		step.SetName(ctn.Name)
-		step.SetNumber(ctn.Number)
-		step.SetStatus(constants.StatusPending)
-		step.SetHost(ctn.Environment["VELA_HOST"])
-		step.SetRuntime(ctn.Environment["VELA_RUNTIME"])
-		step.SetDistribution(ctn.Environment["VELA_DISTRIBUTION"])
+		secret = new(library.Step)
```
Comment (matt-fevold) [inline executor/linux/secret.go:115]: should this be `s`? Doesn't seem like a secret

Comment (jbrockopp) [reply executor/linux/secret.go:115]: The reason it isn't `s` is due to the function header where `s` is already taken:

https://github.com/go-vela/pkg-executor/blob/802c16e3d11aa696727437394ca8e0b7998c605c/executor/linux/secret.go#L105

In regards to not seeming like a secret, I would agree based off the code 👍 

But, this code is responsible for executing the containers for external secrets so it is kind of a gray area.

Comment (matt-fevold) [reply executor/linux/secret.go:115]: 👍 it just looked different than similar sections of code!

**Review by matt-fevold — APPROVED** [review]
LGTM 🐕

**Review by matt-fevold — COMMENTED** [review]
I notice this PR seems to duplicate some of the code in #83 for step load and loadlogs - is that because you wanted to break up the size of each PR and had to add the functions to both PR?

**Comment by jbrockopp** [comment]
@matt-fevold If I'm not mistaken, I don't believe these PRs contain duplicate code.

I could be missing it, but this PR specifically is for adding a `Load()` and `LoadLogs()` functions for the `step` resource.


https://github.com/go-vela/pkg-executor/pull/83 is for adding a `Load()` and `LoadLogs()` functions for the `service` resource.

**Comment by matt-fevold** [comment]
@jbrockopp you're totally right - missed the step/service diff.

File: executor/linux/secret.go [hunk 4]
```diff
@@ -336,7 +341,8 @@ func (s *secretSvc) stream(ctx context.Context, ctn *pipeline.Container) error {
 	r := s.client.repo
 
 	// stream all the logs to the init step
-	l, err := s.client.loadStepLogs(s.client.init.ID)
+	l, err := step.LoadLogs(s.client.init, &s.client.stepLogs)
+	// l, err := s.client.loadStepLogs(s.client.init.ID)
```
Comment (kneal) [inline executor/linux/secret.go:345]: Can this be deleted?

Comment (jbrockopp) [reply executor/linux/secret.go:345]: :doh: Yes you are spot on! 👍 

Sorry about that

Comment (jbrockopp) [reply executor/linux/secret.go]: Fixed 👍

**Review by wass3r — APPROVED** [review]

**Review by kneal — APPROVED** [review]
LGTM 🐬

**Pull request merged**